messaging/src/main/java/org/axonframework/messaging/annotation/MessageHandlerInterceptorDefinition.java

```diff
@@ -89,6 +92,24 @@ public Object handle(Message<?> message, T target) throws Exception {
                     throw e;
                 });
             }
+            finally {
+                if (result instanceof CompletableFuture){
+                    // Invoke the exception handler in case of an exception
+                    ((CompletableFuture<?>) result).exceptionally(exception -> {
+                        try {
+                            ResultParameterResolverFactory.callWithResult(exception, () -> {
+                                if (super.canHandle(message)){
+                                    return super.handle(message, target);
+                                }
+                                return null;
```

Comment:
This will make it so that the information of the exception is lost unless there's a handler and will never reach the gateway client.
It should return the exception to match the behaviour that is not within a CompletableFuture but not as a regular successful return value but as a CompletableFuture that is exceptional.